src/averageAges.js

```diff
@@ -38,6 +55,22 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  const females = people.filter(person => person.sex === 'f');
+  const sumOfAge = females.reduce((totalAge, currentFemale) => {
+    return totalAge + (currentFemale.died - currentFemale.born);
+  }, 0);
+
+  const femalesWithChildren = females.filter(person => {
+    return people.some(child => child.mother === person.name);
+  });
+
+  const ageWithChild = femalesWithChildren.reduce((totalAge, currentFemale) => {
+    return totalAge + (currentFemale.died - currentFemale.born);
+  }, 0);
+
+  return withChildren
+    ? ageWithChild / femalesWithChildren.length
+    : sumOfAge / females.length;
```

Comment:
name the variables so that they mean the same thing
ageWithChild is also sumOfAge, but for different group